File: backend/src/main/java/com/java/NBE4_5_1_7/domain/interviewComment/repository/InterviewCommentRepository.java
package com.java.NBE4_5_1_7.domain.interviewComment.repository;

import java.util.List;
import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import com.java.NBE4_5_1_7.domain.interview.entity.InterviewCategory;
import com.java.NBE4_5_1_7.domain.interviewComment.entity.InterviewContentComment;

import com.java.NBE4_5_1_7.domain.member.entity.Member;

public interface InterviewCommentRepository extends JpaRepository<InterviewContentComment, Long> {
	@Query("SELECT c FROM InterviewContentComment c JOIN c.interviewContent ic " +
		"WHERE c.member = :member AND ic.category = :category")
	List<InterviewContentComment> findByMemberAndInterviewContentCategory(@Param("member") Member member, @Param("category") InterviewCategory category);

	Optional<InterviewContentComment> findById(Long id);

	@Query("select c from InterviewContentComment c where c.interviewContent.interview_content_id = :interviewContentId")
	List<InterviewContentComment> findByInterviewContentId(@Param("interviewContentId") Long interviewContentId);
}

File: backend/src/main/java/com/java/NBE4_5_3_7/domain/interviewComment/repository/InterviewCommentRepository.kt
package com.java.NBE4_5_3_7.domain.interviewComment.repository

import com.java.NBE4_5_3_7.domain.interview.entity.InterviewCategory
import com.java.NBE4_5_3_7.domain.interviewComment.entity.InterviewContentComment
import com.java.NBE4_5_3_7.domain.member.entity.Member
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

interface InterviewCommentRepository : JpaRepository<InterviewContentComment, Long> {

    @Query(
        """
        SELECT c FROM InterviewContentComment c 
        JOIN c.interviewContent ic 
        WHERE c.member = :member AND ic.category = :category
        """
    )
    fun findByMemberAndInterviewContentCategory(
        @Param("member") member: Member,
        @Param("category") category: InterviewCategory
    ): List<InterviewContentComment>

    @Query(
        "SELECT c FROM InterviewContentComment c WHERE c.interviewContent.interviewContentId = :interviewContentId"
    )
    fun findByInterviewContentId(
        @Param("interviewContentId") interviewContentId: Long
    ): List<InterviewContentComment>
}